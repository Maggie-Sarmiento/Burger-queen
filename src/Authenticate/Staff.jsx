import React from "react";
import { useNavigate } from "react-router-dom";
import { auth } from "../Firebase/firebase.config";
import { signOut } from "firebase/auth";
import Sidebar from "../Sidebar";
import styles from './Staff.module.css'

const Staff = () => {
  const navigate = useNavigate();

  const handleLogOut = async () => {
    await signOut(auth);
    navigate('/');
  }

  return (
    <section className={styles.staff}>
      <section>
        <Sidebar />
      </section>
      <header className={styles.header}>
        <button className={styles.btnLogOut} onClick={handleLogOut}>Log Out</button>
      </header>
    </section>
  );
}

export default Staff;
