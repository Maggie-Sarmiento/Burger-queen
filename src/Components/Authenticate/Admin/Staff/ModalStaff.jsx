/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import { useCreateUserWithEmailAndPassword } from 'react-firebase-hooks/auth';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Modal from '@mui/material/Modal';
import TextField from '@mui/material/TextField';
import MenuItem from '@mui/material/MenuItem';
import { setDoc, doc } from 'firebase/firestore';
import { updateCurrentUser, updateProfile } from 'firebase/auth';
import { auth, db } from '../../../../Firebase/firebase.config';
import styles from './ModalStaff.module.css';

const roles = [
  {
    value: 'Admin',
    label: 'Admin',
  },
  {
    value: 'Mesero',
    label: 'Mesero',
  },
  {
    value: 'Jefe de cocina',
    label: 'Jefe de cocina',
  },
];

const style = {
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'background.paper',
  border: '1px solid #000',
  borderRadius: 1,
  boxShadow: 24,
  p: 4,
  textAlign: 'center',
};

const styleT = {
  marginBottom: 1,
};

const ModalStaff = ({ open, setOpen }) => {
  // It does not matter that it has no info, as long as it is an object
  const [data, setData] = useState({});
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confPass, setConfPass] = useState('');

  const [
    createUserWithEmailAndPassword,
    loading,
    error,
  ] = useCreateUserWithEmailAndPassword(auth);

  if (error) {
    return (
      <div>
        <p>
          Error:
          {error.message}
        </p>
      </div>
    );
  }

  if (loading) {
    return console.log('Loading...');
  }

  const handleClose = () => setOpen(false);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setData((preState) => ({
      ...preState,
      [name]: value,
    }));
  };

  const saveDataFirebase = async (localData, id) => {
    await setDoc(doc(db, 'profile', id), localData);
  };

  const saveDataApi = (localData) => {
    const requestOption = {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(localData),
    };

    fetch('http://localhost:3001/empleados', requestOption)
      .then((response) => response.json())
      .catch((err) => console.log(err));
  };

  const handleClick = async (e) => {
    e.preventDefault();
    const expEmail = /^\w+([.+-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,4})+$/;
    const expPassword = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$@$!%*?&])([A-Za-z\d$@$!%*?&]|[^ ]){8,15}$/;
    const alertEmailR = document.getElementById('alertEmailR');

    if (expEmail.test(email) && expPassword.test(password) && expPassword.test(confPass)) {
      if (password === confPass) {
        const originalUser = auth.currentUser; // keeps info of original user to be used later

        await createUserWithEmailAndPassword(email, password); // creates new user
        const newUser = auth.currentUser;
        const { uid } = newUser; // gets new user uid

        const userName = `${data.name} ${data.lastname}`;

        await updateProfile(originalUser, {
          displayName: userName,
        });
        await updateCurrentUser(auth, originalUser); // returns user to original loged in user

        await saveDataFirebase(data, uid); // saves info of created user
        saveDataApi(data);

        handleClose();
      } else {
        alertEmailR.innerHTML = '<span className="red"> Contrase単as Invalida </span>';
      }
    } else {
      alertEmailR.innerHTML = '<span className="red"> Error correo o contrase単a invalida</span>';
    }
  };

  return (
    <section className={styles.modalStaff}>
      <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <form>
          <Box sx={style}>
            <Typography id="modal-modal-title" variant="h6" component="h2">
              Registrar empleado
            </Typography>
            <TextField
              id="date"
              name="date"
              label="Fecha de inicio"
              type="date"
              sx={{ width: 220 }}
              onChange={handleChange}
              InputLabelProps={{
                shrink: true,
              }}
            />
            <TextField
              id="standard-select-currency"
              select
              name="role"
              label="Roles"
              value={data.role}
              onChange={handleChange}
              variant="standard"
            >
              {roles.map((option) => (
                <MenuItem key={option.value} value={option.value}>
                  {option.label}
                </MenuItem>
              ))}
            </TextField>
            <TextField
              required
              id="standard-required"
              name="name"
              label="Nombre"
              variant="standard"
              onChange={handleChange}
              // helperText={textError}
            />
            <TextField
              required
              id="standard-required"
              name="lastname"
              label="Apellido"
              variant="standard"
              onChange={handleChange}
            />
            <TextField
              id="standard-email-input"
              label="Email"
              name="email"
              type="email"
              autoComplete="current-email"
              variant="standard"
              onChange={(e) => {
                setEmail(e.target.value);
                handleChange(e);
              }}
            />
            <TextField
              id="standard-password-input"
              label="Contrase単a"
              type="password"
              autoComplete="current-password"
              variant="standard"
              onChange={(e) => setPassword(e.target.value)}
            />
            <TextField
              id="standard-password-input"
              label="Confirmar contrase単a"
              type="password"
              autoComplete="current-password"
              variant="standard"
              sx={styleT}
              onChange={(e) => setConfPass(e.target.value)}
            />
            <Button onClick={handleClick} type="submit">Guardar</Button>
            <span
              id="alertEmailR"
              className={style.red}
            />
          </Box>
        </form>
      </Modal>
    </section>
  );
};

export default ModalStaff;
