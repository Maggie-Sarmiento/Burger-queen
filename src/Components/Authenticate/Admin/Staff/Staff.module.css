.staff {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 80vh;
  margin-top: 9em;
}

.main {
  display:flex;
}

@media (min-width: 768px) {
  .main {
    display:flex;
    justify-content: center;
    margin-top: 1em;
    padding: 0 2em;
  }
  .staffBtnContainer {
    margin-left: 2.2em;
    margin-top: 1em;
  }
}

@media (min-width: 1000px) {
  .staff {
    margin-top: 6em;
  }
}

@media (min-width: 1280px) {
  .main {
    display:flex;
    justify-content: center;
    margin-top: 1em;
    padding: 0 5em;
  }
}
