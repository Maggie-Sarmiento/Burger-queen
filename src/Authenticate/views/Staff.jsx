import React, { useEffect, useState } from 'react';
// import { makeStyles } from '@material-ui/core/styles';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Modal from '@mui/material/Modal';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import ModalStaff from '../Components/ModalStaff';
import styles from './Staff.module.css';

const Staff = () => {
  const [dataStaff, setDataStaff] = useState([]);
  const [modalAdd, setModalAdd] = useState(false);
  const [modalEdit, setModalEdit] = useState(false);
  const [modalDelete, setModalDelete] = useState(false);

  const openCLoseModalAdd = () => {
    setModalAdd(!modalAdd);
  };

  const openCLoseModalEdit = () => {
    setModalEdit(!modalEdit);
  };

  const openCLoseModalDelete = () => {
    setModalDelete(!modalDelete);
  };

  useEffect(() => {
    fetch('http://localhost:3001/empleados')
      .then((response) => response.json())
      .then((data) => setDataStaff(data));
  }, []);

  // console.log(dataStaff);

  const bodyAdd = (
    <div>
      <h3>Agregar Empleado</h3>
      <TextField name="nombre" label="Nombre" />
      <br />
      <TextField name="apellido" label="Apellido" />
      <br />
      <TextField name="rol" label="Rol" />
      <br />
      <div align="right">
        <Button color="primary">Insertar</Button>
        <Button>Cancelar</Button>
      </div>
    </div>
  );

  const bodyEdit = (
    <div>
      <h3>Editar</h3>
      <TextField name="nombre" label="Nombre" />
      <br />
      <TextField name="apellido" label="Apellido" />
      <br />
      <TextField name="rol" label="Rol" />
      <br />
      <TextField name="fecha-de-inicio" label="Fecha de Inicio" />
      <br />
      <TextField name="email" label="Email" />
      <br />
      <div align="right">
        <Button color="primary">Editar</Button>
        <Button>Cancelar</Button>
      </div>
    </div>
  );

  const bodyDelete = (
    <div>
      <p>
        Estás seguro que deseas eliminar la consola
      </p>
      <div align="right">
        <Button color="secondary">Sí</Button>
        <Button onClick={() => openCLoseModalDelete()}>No</Button>
      </div>

    </div>
  );

  return (
    <section className={styles.staff}>
      <ModalStaff />
      <main className={styles.main}>
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Id</TableCell>
                <TableCell>Nombre</TableCell>
                <TableCell>Apellido</TableCell>
                <TableCell>Rol</TableCell>
                <TableCell>Correo</TableCell>
                <TableCell>Fecha de inicio</TableCell>
                <TableCell>Acciones</TableCell>
              </TableRow>
            </TableHead>

            <TableBody>
              {dataStaff.map((empleado) => (
                <TableRow key={empleado.id}>
                  <TableCell>{empleado.id}</TableCell>
                  <TableCell>{empleado.nombre}</TableCell>
                  <TableCell>{empleado.apellido}</TableCell>
                  <TableCell>{empleado.rol}</TableCell>
                  <TableCell>{empleado.correo}</TableCell>
                  <TableCell>{empleado.fecha}</TableCell>
                  <TableCell>
                    <EditIcon onClick={openCLoseModalEdit} />
                    &nbsp;&nbsp;&nbsp;
                    <DeleteIcon onClick={openCLoseModalDelete} />
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>

        <Modal
          open={modalAdd}
          onClose={openCLoseModalAdd}
        >
          {bodyAdd}
        </Modal>

        <Modal
          open={modalEdit}
          onClose={openCLoseModalEdit}
        >
          {bodyEdit}
        </Modal>

        <Modal
          open={modalDelete}
          onClose={openCLoseModalDelete}
        >
          {bodyDelete}
        </Modal>
      </main>
    </section>
  );
};

export default Staff;
