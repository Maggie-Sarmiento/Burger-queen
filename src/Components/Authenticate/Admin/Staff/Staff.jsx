/* eslint-disable no-underscore-dangle */
/* eslint-disable no-unused-expressions */
import React, { useEffect, useState } from 'react';
// import { makeStyles } from '@material-ui/core/styles';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Modal from '@mui/material/Modal';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import MenuItem from '@mui/material/MenuItem';
import Box from '@mui/material/Box';
import ModalStaff from './ModalStaff/ModalStaff';
import StaffBtn from './StaffBtn';
import styles from './Staff.module.css';
// import { flexbox } from '@mui/system';

const modal = {
  display: 'flex',
  flexDirection: 'column',
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: 400,
  bgcolor: 'background.paper',
  border: '1px solid #000',
  borderRadius: 1,
  boxShadow: 24,
  p: 4,
  textAlign: 'center',
};

const roles = [
  {
    value: 'Admin',
    label: 'Admin',
  },
  {
    value: 'Mesero',
    label: 'Mesero',
  },
  {
    value: 'Jefe de cocina',
    label: 'Jefe de cocina',
  },
];

const Staff = () => {
  const [dataStaff, setDataStaff] = useState([]);
  const [modalEdit, setModalEdit] = useState(false);
  const [modalDelete, setModalDelete] = useState(false);
  const [open, setOpen] = useState(false);
  const [userEdit, setUserEdit] = useState(null);

  const openCLoseModalEdit = () => {
    setModalEdit(!modalEdit);
  };

  const openCLoseModalDelete = () => {
    setModalDelete(!modalDelete);
  };

  const getStaff = (empleado, action) => {
    setUserEdit(empleado);
    (action === 'Edit') ? openCLoseModalEdit() : openCLoseModalDelete();
  };

  const handleOpen = () => setOpen(true);

  const editDataApi = (e, id) => {
    const data = e.target.form;

    const employeeData = {
      date: data[0].value,
      role: data[2].value,
      name: data[3].value,
      lastname: data[4].value,
      email: data[5].value,
    };

    const requestOption = {
      method: 'PUT',
      headers: { 'Content-Type': 'application/json', Authorization: localStorage.getItem('token') },
      body: JSON.stringify(employeeData),
    };

    fetch(`http://localhost:8080/staffs/${id}`, requestOption)
      .then((response) => {
        response.json();
        employeeData.id = id;
        setDataStaff([...dataStaff.filter((staff) => staff.id !== id), employeeData]);
      })
      .catch((err) => err);
    openCLoseModalEdit();
  };

  const deleteDataApi = (id) => {
    const requestOption = {
      method: 'DELETE',
      headers: { 'Content-Type': 'application/json', Authorization: localStorage.getItem('token') },
    };

    fetch(`http://localhost:8080/staffs/${id}`, requestOption)
      .then((response) => response.json())
      .then(() => {
        setDataStaff(dataStaff.filter((staff) => staff.id !== userEdit.id));
      })
      .catch((err) => err);
    openCLoseModalDelete();
  };

  const bodyEdit = (
    <form>
      <h3>Editar</h3>
      <TextField
        id="date"
        name="date"
        label="Fecha de inicio"
        type="date"
        sx={{ width: 300 }}
        defaultValue={userEdit && userEdit.date}
        InputLabelProps={{
          shrink: true,
        }}
      />
      <br />
      <TextField
        id="standard-select-currency"
        select
        name="role"
        label="Roles"
        defaultValue={userEdit && userEdit.role}
        variant="standard"
        sx={{ width: 300 }}
      >
        {roles.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
      </TextField>
      <br />
      <TextField
        required
        id="standard-required"
        name="name"
        label="Nombre"
        variant="standard"
        defaultValue={userEdit && userEdit.name}
        sx={{ width: 300 }}
      />
      <br />
      <TextField
        required
        id="standard-required"
        name="lastname"
        label="Apellido"
        variant="standard"
        defaultValue={userEdit && userEdit.lastname}
        sx={{ width: 300 }}
      />
      <br />
      <TextField
        id="standard-email-input"
        label="Email"
        name="email"
        type="email"
        autoComplete="current-email"
        variant="standard"
        defaultValue={userEdit && userEdit.email}
        sx={{ width: 300 }}
      />
      <br />
      <br />
      <div align="right">
        <Button type="submit" color="primary" onClick={(e) => editDataApi(e, userEdit._id)}>Guardar</Button>
        <Button type="button" onClick={() => openCLoseModalEdit()}>Cancelar</Button>
      </div>
    </form>
  );

  useEffect(() => {
    const requestOption = {
      method: 'GET',
      headers: { 'Content-Type': 'application/json', Authorization: localStorage.getItem('token') },
    };
    fetch('http://localhost:8080/staffs', requestOption)
      .then((response) => response.json())
      .then((data) => setDataStaff(data));
  }, []);

  return (
    <section className={styles.staff}>
      <div className={styles.staffBtnContainer}>
        <StaffBtn handleClick={handleOpen} />
      </div>
      <div className={styles.modalCar}>
        <ModalStaff open={open} setOpen={setOpen} />
      </div>
      <main className={styles.main}>
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell><b>Nombre</b></TableCell>
                <TableCell><b>Apellido</b></TableCell>
                <TableCell><b>Rol</b></TableCell>
                <TableCell><b>Correo</b></TableCell>
                <TableCell><b>Fecha de Inicio</b></TableCell>
                <TableCell><b>Acciones</b></TableCell>
              </TableRow>
            </TableHead>

            <TableBody>
              {dataStaff.map((empleado) => (
                <TableRow key={empleado._id}>
                  <TableCell>{empleado.name}</TableCell>
                  <TableCell>{empleado.lastname}</TableCell>
                  <TableCell>{empleado.role}</TableCell>
                  <TableCell>{empleado.email}</TableCell>
                  <TableCell>{empleado.date}</TableCell>
                  <TableCell>
                    <EditIcon onClick={() => getStaff(empleado, 'Edit')} />
                    &nbsp;&nbsp;&nbsp;
                    <DeleteIcon onClick={() => getStaff(empleado, 'Delete')} />
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>

        <Modal
          open={modalEdit}
          onClose={openCLoseModalEdit}
        >
          <Box sx={modal}>
            {bodyEdit}
          </Box>
        </Modal>

        <Modal
          open={modalDelete}
          onClose={openCLoseModalDelete}
        >
          <Box sx={modal}>
            <div>
              <p>
                Estás seguro que deseas eliminar a
                <br />
                <span>{userEdit && userEdit.name}</span>
                ?
              </p>
              <div align="right">
                <Button color="secondary" onClick={() => deleteDataApi(userEdit._id)}>Sí</Button>
                <Button onClick={() => openCLoseModalDelete()}>No</Button>
              </div>
            </div>
          </Box>
        </Modal>
      </main>
    </section>
  );
};

export default Staff;
