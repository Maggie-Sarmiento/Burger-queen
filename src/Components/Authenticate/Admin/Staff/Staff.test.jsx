/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import Staff from './Staff';

describe('test Staff', () => {
  test('render component Staff', async () => {
    let root;
    act(() => {
      root = create(
        <Router>
          <Staff />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
