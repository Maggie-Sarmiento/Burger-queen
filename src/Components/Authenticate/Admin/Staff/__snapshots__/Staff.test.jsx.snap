// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Staff render Staff 1`] = `
<section
  className="staff"
>
  <div
    className="staffBtnContainer"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-zc09lo-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      Empleado
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    className="modalCar"
  >
    <section
      className="modalStaff"
    />
  </div>
  <main
    className="main"
  >
    <div
      className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    >
      <table
        className="MuiTable-root css-rqglhn-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Nombre
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Apellido
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Rol
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Correo
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Fecha de Inicio
              </b>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              <b>
                Acciones
              </b>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </main>
</section>
`;
